webapp/security/views.py

```diff
@@ -354,6 +354,15 @@ def cve_index():
         .all()
     )
 
+    cve_tree = defaultdict(dict)
+    for cve in cves:
+        packages = cve.status_tree
+
+        if package:
+            packages = {k: v for k, v in packages.items() if k == package}
+
+        cve_tree[cve.id] = packages
```

Comment:
Ideally we should do this in the original cves_query, it could be expensive to loop through thousands of packages, @jpmartinspt any ideas?

Reply:
I might be misreading the purpose of this.. But if is to only show packages on the template that match this filters we could just add a couple of `if` statmentes on the template itself since we already loop through the packages there to render them?

Reply:
In the appropriate place in `_cve-table.html` -> `{% if request.args.get('package') and package_name = request.args.get('package' ) %}` so we only render packages with the matching name.

Reply:
Filtering in the packages in the view would be a nice way to avoid some extra for loops. The only problem is that there are more filters to be applied fx: filtering by component in my second PR: https://github.com/canonical-web-and-design/ubuntu.com/pull/8006 The view would end up containing too much logic. 

I do agree that it would be the cleanest to make `cves_query` do the filtering, but right now, I am not sure how..

Reply:
Two other oprions:
1) Transform each `cve` into a dict instead of just passing down the sqlalchemy object to the template. That way we can change and filter the packages field.(can be tricky if we are using sqlalchemy specific stuff in the template logic)

2) Make `cve.status_tree` a `hybrid_method` that takes arguments so we can filter down the tree to whatever we want.

Reply:
1. Would this not ruin the pagination which is implemented with `sqlalchemy`'s help (using `offset` and `limit`)?
2. This is interesting. Would you pass `request.args.get('package')` as a parameter inside the `hybrid_method`?

Reply:
For the second option, yes you would call the method on the template with arguments.

Reply:
I found a way to make it work using `cves_query`. @carkod could you please check the code one more time. It's much cleaner now.